vermouth/tests/test_molecule.py

```diff
@@ -1279,3 +1279,49 @@ def test_to_molecule():
                              parameters=['a', '0.1', '300'],
                              meta={'b': 1}),]
     assert ref_bonds == test_molecule.interactions['bonds']
+
+
+@pytest.mark.parametrize('atoms, bonds, interactions, removed, expected', [
+    # empty molecule
+    ([], [], [], {}, {}),
+    # interactions that all need to be removed
+    ([1, 2], [(1, 2)], [('bond', (1, 2), {})], {'type_': 'bond', "atoms": (1, 2)}, {}),
+    # Molecule with interactions of different types of which some need to be removed
```

Comment:
to future proof this I would add a test with populated meta and parameter dict. Just to make sure if someone touches it in future the test will fail.